core/parse.ml

```diff
@@ -8,28 +8,35 @@ open Lexing
    occurs.
 *)
 
-type 'a grammar = (Lexing.lexbuf -> Parser.token) -> Lexing.lexbuf -> 'a
-
 class source_code = SourceCode.source_code
 
-(* Read and parse Links source code from the source named `name' via
-   the function `infun'.
-*)
-let read : context:Lexer.lexer_context
-        -> ?nlhook:(unit -> unit)
-        -> parse:('intermediate grammar)
-        -> infun:(bytes -> int -> int)
-        -> name:string
-        -> 'result * source_code =
-fun ~context ?nlhook ~parse ~infun ~name ->
-  let code = new source_code in
-  let lexbuf = {(from_function (code#parse_into infun))
+module type LexerSig = sig
```

Comment:
From here to line 79, I'm not sure why this code should be in parse.ml.  Since it is reused in parseXml.ml which does not otherwise need anything from parse.ml, I suggest it may be moved to a standalone file, perhaps something like `lex.ml` or `lexerIntf.ml`.

Reply:
Frankly speaking, I had similar thoughts. I'll put it in a separate file.